Before:

```
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error : The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Shared.ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile(Boolean condition, String errorSubCategoryResourceName, BuildEventFileInfo projectFile, Exception innerException, String resourceName, Object[] args) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Shared.ProjectFileErrorUtilities.ThrowInvalidProjectFile(BuildEventFileInfo projectFile, Exception innerException, String resourceName, Object[] args) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Construction.ProjectRootElement.LoadDocument(String fullPath, Boolean preserveFormatting, Boolean loadAsReadOnly) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Construction.ProjectRootElement..ctor(String path, ProjectRootElementCacheBase projectRootElementCache, Boolean preserveFormatting) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Construction.ProjectRootElement.CreateProjectFromPath(String projectFile, ProjectRootElementCacheBase projectRootElementCache, Boolean preserveFormatting) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Evaluation.ProjectRootElementCache.Get(String projectFile, OpenProjectRootElement loadProjectRootElement, Boolean isExplicitlyLoaded, Nullable`1 preserveFormatting) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Construction.ProjectRootElement.OpenProjectOrSolution(String fullPath, IDictionary`2 globalProperties, String toolsVersion, ProjectRootElementCacheBase projectRootElementCache, Boolean isExplicitlyLoaded) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Execution.ProjectInstance..ctor(String projectFile, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, Nullable`1 projectLoadSettings, EvaluationContext evaluationContext) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Execution.ProjectInstance.FromFile(String file, ProjectOptions options) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at NuGet.Build.Tasks.Console.MSBuildStaticGraphRestore.<>c__DisplayClass38_1.<LoadProjects>b__0(String path, Dictionary`2 properties, ProjectCollection collection) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.GraphBuilder.ParseProject(ConfigurationMetadata configurationMetadata) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at System.Lazy`1.CreateValue() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at System.Lazy`1.LazyInitValue() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.ParallelWorkSet`2.ExecuteWorkItem() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.ParallelWorkSet`2.WaitForAllWorkAndComplete() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.GraphBuilder.FindGraphNodes() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.GraphBuilder.BuildGraph() [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at Microsoft.Build.Graph.ProjectGraph..ctor(IEnumerable`1 entryPoints, ProjectCollection projectCollection, ProjectInstanceFactoryFunc projectInstanceFactory, Int32 degreeOfParallelism, CancellationToken cancellationToken) [<PATHTOPROJECT>]
C:\Program Files\Visual Studio 2022\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(19,5): error :    at NuGet.Build.Tasks.Console.MSBuildStaticGraphRestore.LoadProjects(IEnumerable`1 entryProjects) [<PATHTOPROJECT>]
    0 Warning(s)
    1 Error(s)
```

After:
```
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
<PATHTOPROJECT> : error MSB4025: The project file could not be loaded. Could not find a part of the path '<PATHTOPROJECT>'.  <PATHTOPROJECT> [<PATHTOPROJECT>]
    0 Warning(s)
    14 Error(s)
```

/cc @AArnott

--- a/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
+++ b/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
@@ -766,7 +766,7 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
         }
 
         [PlatformFact(Platform.Windows)]
-        public void MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileExceptionn()
+        public void MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileException()
         {
             // Arrange
             using (var pathContext = new SimpleTestPathContext())
@@ -779,12 +779,10 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
                 var projectA = new SimpleTestProjectContext("a", ProjectStyle.PackageReference, pathContext.SolutionRoot);
 
                 var projectB = new SimpleTestProjectContext("b", ProjectStyle.PackageReference, pathContext.SolutionRoot);
-                var projectC = new SimpleTestProjectContext("c", ProjectStyle.PackageReference, pathContext.SolutionRoot);
 
                 var projectAFrameworkContext = new SimpleTestProjectFrameworkContext(net461);
 
                 projectAFrameworkContext.ProjectReferences.Add(projectB);
-                projectAFrameworkContext.ProjectReferences.Add(projectC);
 
                 var packageX = new SimpleTestPackageContext()
                 {
@@ -799,16 +797,13 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
                 solution.Create(pathContext.SolutionRoot);
 
                 File.Delete(projectB.ProjectPath);
-                File.Delete(projectC.ProjectPath);
 
-                // Restore the project with a PackageReference which generates assets
                 var result = _msbuildFixture.RunMsBuild(pathContext.WorkingDirectory, $"/t:restore /p:RestoreUseStaticGraphEvaluation=true {projectA.ProjectPath}", ignoreExitCode: true);
 
                 // Assert
                 Assert.True(result.ExitCode == 1, result.AllOutput);
 
                 result.AllOutput.Should().Contain($"error MSB4025: The project file could not be loaded. Could not find file '{projectB.ProjectPath}'");
-                result.AllOutput.Should().Contain($"error MSB4025: The project file could not be loaded. Could not find file '{projectC.ProjectPath}'");
             }
         }
 

--- a/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
+++ b/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
@@ -766,7 +766,7 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
         }
 
         [PlatformFact(Platform.Windows)]
-        public async Task MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileExceptionn()
+        public void MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileExceptionn()
         {
             // Arrange
             using (var pathContext = new SimpleTestPathContext())
@@ -793,29 +793,10 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
                 };
                 projectA.Frameworks.Add(projectAFrameworkContext);
 
-                packageX.Files.Clear();
                 projectA.AddPackageToAllFrameworks(packageX);
-                packageX.AddFile("lib/net461/a.dll");
 
                 solution.Projects.Add(projectA);
                 solution.Create(pathContext.SolutionRoot);
-
-                var configAPath = Path.Combine(Path.GetDirectoryName(projectA.ProjectPath), "NuGet.Config");
-                var configText =
-$@"<?xml version=""1.0"" encoding=""utf-8""?>
-<configuration>
-    <packageSources>
-        <add key=""LocalSource"" value=""{pathContext.PackageSource}"" />
-    </packageSources>
-</configuration>";
-                using (var writer = new StreamWriter(configAPath))
-                {
-                    writer.Write(configText);
-                }
-
-                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
-                    pathContext.PackageSource,
-                    packageX);
 
                 File.Delete(projectB.ProjectPath);
                 File.Delete(projectC.ProjectPath);

--- a/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
+++ b/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs
@@ -765,6 +765,72 @@ $@"<?xml version=""1.0"" encoding=""utf-8""?>
             return Task.CompletedTask;
         }
 
+        [PlatformFact(Platform.Windows)]
+        public async Task MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileExceptionn()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                var projectA = new SimpleTestProjectContext("a", ProjectStyle.PackageReference, pathContext.SolutionRoot);
+
+                var projectB = new SimpleTestProjectContext("b", ProjectStyle.PackageReference, pathContext.SolutionRoot);
+                var projectC = new SimpleTestProjectContext("c", ProjectStyle.PackageReference, pathContext.SolutionRoot);
+
+                var projectAFrameworkContext = new SimpleTestProjectFrameworkContext(net461);
+
+                projectAFrameworkContext.ProjectReferences.Add(projectB);
+                projectAFrameworkContext.ProjectReferences.Add(projectC);
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+                projectA.Frameworks.Add(projectAFrameworkContext);
+
+                packageX.Files.Clear();
+                projectA.AddPackageToAllFrameworks(packageX);
+                packageX.AddFile("lib/net461/a.dll");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                var configAPath = Path.Combine(Path.GetDirectoryName(projectA.ProjectPath), "NuGet.Config");
+                var configText =
+$@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+        <add key=""LocalSource"" value=""{pathContext.PackageSource}"" />
+    </packageSources>
+</configuration>";
+                using (var writer = new StreamWriter(configAPath))
+                {
+                    writer.Write(configText);
+                }
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    packageX);
+
+                File.Delete(projectB.ProjectPath);
+                File.Delete(projectC.ProjectPath);
+
+                // Restore the project with a PackageReference which generates assets
+                var result = _msbuildFixture.RunMsBuild(pathContext.WorkingDirectory, $"/t:restore /p:RestoreUseStaticGraphEvaluation=true {projectA.ProjectPath}", ignoreExitCode: true);
+
+                // Assert
+                Assert.True(result.ExitCode == 1, result.AllOutput);
+
+                result.AllOutput.Should().Contain($"error MSB4025: The project file could not be loaded. Could not find file '{projectB.ProjectPath}'");
+                result.AllOutput.Should().Contain($"error MSB4025: The project file could not be loaded. Could not find file '{projectC.ProjectPath}'");
+            }
+        }
+
         [PlatformTheory(Platform.Windows)]
         [InlineData(true)]
         [InlineData(false)]

